source/importers/SynchronizeColorsTokens.ts

```diff
@@ -0,0 +1,218 @@
+	}
+
+	return hexToCssVariable;
+};
+
+const ImportSingleTokenFile = (
+	binValues : configFileObject,
+	jsonColors: Array<any>,
+	hexToCssVariable: Object,
+) => {
+	try {
+		let result: Array<string> = [];
+		let resultJson = {};
+
+		jsonColors.forEach(obj => {
+			const patternColorsToIgnore = /deprecated|Pattern|RAW|theme/i;
+			if (obj.name.match(patternColorsToIgnore) === null) {
+				let tokenName = obj.name;
+				tokenName = tokenName.replace(/\//i, '-');
+
+				if (obj.values.hex .split('').every(char => char === obj.values.hex [0])){
+					obj.values.hex  = `${obj.values.hex [0]}${obj.values.hex [1]}${obj.values.hex [2]}`
+				}
+
+				if (obj.values.alpha !== 1) {
+					result.push('$' + tokenName + ': ' + 'rgba(var(' + hexToCssVariable[obj.values.hex] +'-rgb), ' + obj.values.alpha + ');');
+				} else {
+					result.push('$' + tokenName + ': ' + hexToCssVariable[obj.values.hex] + ';');
+				}
+
+				/** JSON object structure */
+				const tokenNameSplitted = tokenName.split('-');
+				const category = tokenNameSplitted[1];
+				const resultJsonObject: ITokenJsonObject = {
+					id: binValues.destination + '_' + tokenName,
+					label: tokenName,
+					value: hexToCssVariable[obj.values.hex],
+				};
+				if (resultJson[category] === undefined) {
+					resultJson[category] = [];
+				}
+				resultJson[category].push(resultJsonObject);
+			}
+		});
+		arrayToFile(tokensFilesConfig.destinationPath + binValues.destination, result);
+		jsonToFile(tokensFilesConfig.destinationPath + binValues.destinationJson, resultJson);
+	} catch (err) {
+		console.error(err)
+	}
+}
+
+
+const hexToRgb = (hex: string ) => {
+	if (typeof hex !== "string" || [4, 7].includes(hex.length) === false) {
+		return null;
+	}
+
+	if (hex.length === 4) {
+		hex = hex + hex[1] + hex[2] + hex[3];
+	}
+	let result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+	return result ? '' + parseInt(result[1], 16) + ', ' +
+		parseInt(result[2], 16) + ', ' + parseInt(result[3], 16)  : null;
+}
+
+const arrayToFile = (filepath: string, content: Array<string>) => {
+	let file = fsT.createWriteStream(filepath);
+	file.on('error', function(err) { console.log(err) });
+
+	const patternTheme = /root|}/i;
+	let hasFileIndentation = false;
+	content.forEach(function(v) {
+		if (v.match(patternTheme)) {
+			hasFileIndentation = true;
+			file.write( v.toLowerCase() + '\n');
+		} else {
+			const hasLineIndentation = hasFileIndentation ? '\t' : ''
+			file.write( hasLineIndentation+ v.toLowerCase() + '\n');
+		}
+	});
+	file.end();
+}
+
+const jsonToFile  = (filepath: string, content: Object) => {
+	let file = fsT.createWriteStream(filepath);
+	file.on('error', function(err) { console.log(err) });
+	file.write(JSON.stringify(content));
+	file.end();
+}
+
+const SynchronizeSingleBin = async (bin) => {
+	try {
+		const response = await axios.get(tokensFilesConfig.jsonBinApiUrl + bin.id + '/latest');
+		let hexToCssVariable;
+		Object.keys(bin.files).forEach(() => {
```

Comment:
do we need this `forEach()`? Aren't we performing the same operations twice now?

Reply:
True, I'm removing this loop.